dipy/reconst/dki.py

```diff
@@ -0,0 +1,1634 @@
+        L3 = c[cond1]
+        RFm = rfpython(L1/L2, L1/L3, np.ones(len(L1)))
+        RDm = rdpython(L1/L2, L1/L3, np.ones(len(L1)))
+        F1[cond1] = ((L1+L2+L3) ** 2) / (18 * (L1-L2) * (L1-L3)) * \
+                    (((np.sqrt(L2*L3)) / L1) * RFm + \
+                     ((3 * L1**2 - L1*L2 - L1*L3 - L2*L3) / \
+                     (3 * L1 * np.sqrt(L2*L3))) * RDm - 1)
+
+    # Resolve possible sigularity a==b
+    cond2 = np.logical_and(~cond0, np.logical_and(abs(a - b) < er,
+                                                  abs(b - c) > er))
+    if np.sum(cond2)!=0:
+        L1 = a[cond2]
+        L3 = c[cond2]
+        F1[cond2] = F2m(L3, L1, L1) / 2
+
+    # Resolve possible sigularity a==c 
+    cond3 = np.logical_and(~cond0, np.logical_and(abs(a - c) < er,
+                                                  abs(a - b) > er))
+    if np.sum(cond3)!=0:
+        L1 = a[cond3]
+        L2 = b[cond3]
+        F1[cond3] = F2m(L2, L1, L1) / 2
+
+    # Resolve possible sigularity a==b and a==c
+    cond4 = np.logical_and(~cond0, np.logical_and(abs(a - c) < er,
+                                                  abs(a - b) < er))
+    if np.sum(cond4)!=0:
+        F1[cond4] = 1/5.
+
+    return F1
+
+
+def F2m(a,b,c):
+    """
+    Helper function that computes function $F_2$ which is required to compute
+    the analytical solution of the Mean kurtosis.
+    
+    Parameters
+    ----------
+    a : ndarray (...)
+        Array containing the first variable of function $F_1$ for all signal
+        voxels. This variable will be one of the three diffusion tensor
+        eigenvalues.
+    b : ndarray (...)
+        Array containing the second variable of function $F_1$ for all signal
+        voxels. This variable will be one of the three diffusion tensor
+        eigenvalues.
+    c : ndarray (...)
+        Array containing the thrid variable of function $F_1$ for all signal
+        voxels. This variable will be one of the three diffusion tensor
+        eigenvalues.
+        
+    Returns
+    -------
+    F2 : array
+       Value of the function $F_2$ given the parameters a, b, and c
+
+    Notes
+    --------
+    Function F_2 is defined as [1]_:
+
+    \begin{multline}
+    F_2(\lambda_1,\lambda_2,\lambda_3)=
+    \frac{(\lambda_1+\lambda_2+\lambda_3)^2}
+    {3(\lambda_2-\lambda_3)^2}
+    [\frac{\lambda_2+\lambda_3}{\sqrt{\lambda_2\lambda_3}}
+    R_F(\frac{\lambda_1}{\lambda_2},\frac{\lambda_1}{\lambda_3},1)+\\
+    \frac{2\lambda_1-\lambda_2-\lambda_3}{3\sqrt{\lambda_2 \lambda_3}}
+    R_D(\frac{\lambda_1}{\lambda_2},\frac{\lambda_1}{\lambda_3},1)-2]
+    \end{multline}
+
+    References
+    ----------
+    .. [1] Tabesh, A., Jensen, J.H., Ardekani, B.A., Helpern, J.A., 2011.
+           Estimation of tensors and tensor-derived measures in diffusional
+           kurtosis imaging. Magn Reson Med. 65(3), 823-836
+    """
+    # Float error used to compare two floats, abs(l1 - l2) < er for l1 = l2 
+    er = 1e-10
```

Comment:
An easier and more robust way to get the machine epsilon across different architectures for a 
given float type is to use np.finfo():

print(np.finfo(float).eps)
# 2.22044604925e-16

print(np.finfo(np.float32).eps)
# 1.19209e-07

So, 1e-10 that you use here could be not enough for 64bit and too low for 32bit.